py4DSTEM/process/phase/iterative_base_class.py

```diff
@@ -1217,7 +1217,7 @@ def _normalize_diffraction_intensities(
         for rx in range(diffraction_intensities.shape[0]):
             for ry in range(diffraction_intensities.shape[1]):
                 if positions_mask is not None:
-                    if not self._positions_mask[rx, ry]:
+                    if not positions_mask[rx, ry]:
                         continue
```

Comment:
Is this skipping posistions where `posistions_mask[rx,ry] == False` if so is the following correct and more readable"
```python
if posistion_mask[rx,ry] is False:
    continue 
```

Is it worth splitting this into two for loops so that `posistion_mask is Not None`, is checked once, and not per probe position?

Reply:
Correct, this is skipping positions where the mask is false. Not sure I understand the for loop suggestion, this has to be checked per probe position. 

Re: style - PEP8 explicitly says comparing boolean values to True/False with == or is is bad form.

Reply:
Good catch. One day I'll learn to not try and refactor numpy Boolean masks

Reply:
About the for loop, this checks if `posistion_mask is not None` at every probe position.  Would it be better to do ~:

```
 if positions_mask is None:
     intensities = get_shifted_ar(
                    diffraction_intensities[rx, ry],
                    -com_fitted_x[rx, ry],
                    -com_fitted_y[rx, ry],
                    bilinear=True,
                    device="cpu",
                )

                if crop_patterns:
                    intensities = intensities[crop_mask].reshape(
                        region_of_interest_shape
                    )

                mean_intensity += np.sum(intensities)
                amplitudes[counter] = np.sqrt(np.maximum(intensities, 0))
                counter += 1
else:
    for rx in range(diffraction_intensities.shape[0]):
            for ry in range(diffraction_intensities.shape[1]):
                  if not positions_mask[rx, ry]:
                      continue
              intensities = get_shifted_ar(
                  diffraction_intensities[rx, ry],
                  -com_fitted_x[rx, ry],
                  -com_fitted_y[rx, ry],
                  bilinear=True,
                  device="cpu",
              )

              if crop_patterns:
                  intensities = intensities[crop_mask].reshape(
                      region_of_interest_shape
                  )

                mean_intensity += np.sum(intensities)
                amplitudes[counter] = np.sqrt(np.maximum(intensities, 0))
                counter += 1
```